src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -2354,10 +2356,13 @@ bool QgsOgrProvider::_setSubsetString( const QString &theSQL, bool updateFeature
 
   mRefreshFeatureCount = updateFeatureCount;
 
-  // check the validity of the layer
-  QgsDebugMsgLevel( QStringLiteral( "checking validity" ), 4 );
-  loadFields();
-  QgsDebugMsgLevel( QStringLiteral( "Done checking validity" ), 4 );
+  // check the validity of the layer if subset string has changed
+  if ( subsetStringHasChanged )
+  {
+    QgsDebugMsgLevel( QStringLiteral( "checking validity" ), 4 );
```

Comment:
these look like outdated comments now... can they be removed, or does loadFields() **actually** check for validity?